File: MiniEP1/src/main.py
from random import Random


def pi(seed: int, runs: int) -> float:
    """Estimates pi through monte carlo"""
    
    rng = Random(seed)
    hits = 0
    
    for i in range(runs):
        x, y = rng.random(), rng.random()
        if x*x + y*y <= 1.0:
            hits += 1
    
    return 4*hits/runs


if __name__ == '__main__':
    print(pi(2022, 10**7))

File: MiniEP1/src/main.go
package main

import (
	"fmt"
	"math/rand"
)

// Estimates pi through monte carlo
func pi(seed int, runs int) float64 {

	rng := rand.New(rand.NewSource(int64(seed)))
	hits := 0

	for i := 0; i < runs; i++ {
		x, y := rng.Float64(), rng.Float64()
		if x*x+y*y < 1.0 {
			hits++
		}
	}

	return float64(4*hits) / float64(runs)
}

func main() {
	fmt.Println(pi(2022, 1e7))
}
